tomee/tomee-catalina/src/main/java/org/apache/tomee/catalina/TomcatLoader.java

```diff
@@ -187,7 +189,7 @@ public void initialize(final Properties properties) throws Exception {
         System.setProperty("openejb.base", SystemInstance.get().getBase().getDirectory().getAbsolutePath());
 
         // Install tomcat thread context listener
-        ThreadContext.addThreadContextListener(new TomcatThreadContextListener());
+        ThreadContext.addThreadContextListener(threadContextListener);
```

Comment:
Again, not comfortable with static listeners. Is this only instantiated once, globally?

Reply:
Same reason like above for RequestScopedThreadContextListener 

These ThreadContextListeners don't depend on anything else and they just manipulate the passed data.